corehq/apps/reports_core/filters.py

```diff
@@ -465,16 +466,19 @@ def valid_location_ids(self, location_id):
 
     def value(self, **kwargs):
         selected_loc_id = kwargs.get(self.name, None)
+        request_user = kwargs.get(REQUEST_USER_KEY, None)
         if selected_loc_id:
-            return self.valid_location_ids(selected_loc_id)
+            if request_user and selected_loc_id in request_user.get_location_ids(self.domain):
```

Comment:
```suggestion
            if request_user and user_can_access_location_id(self.domain, request_user, selected_loc_id):
```
The user should also be able to access any descendant locations of their assigned locations.  That handler should take care of that.